### PR TITLE
Add tasklist HUD

### PR DESCRIPTION
**Pull request by hirschenberger**
PR open for discussion. See #236

### CODE REVIEW

**Comment by wereturtle** [comment]
Hi @hirschenberger!

It appears there is a bug.  I had just compiled your code and began to type in a new task list item.  Unfortunately, while typing the first few characters after manually typing `- [ ]`, it put the cursor at the beginning of the paragraph and began inserting characters there.

Alas, I do not have the time to debug this.  Happy bug hunting!

**Comment by hirschenberger** [comment]
Sorry, I added a bug with the latest version. I updated the PR, you can check it now.

**Comment by hirschenberger** [comment]
@wereturtle Should I open a new PR for this?

**Comment by wereturtle** [comment]
Hi @hirschenberger, I spent some time today testing the code on your branch.  Here are a few issues I found:

1. It's not making the next task list item when I press `Enter` in the middle of a list.  This was the original functionality.  It needs to be put back in.
2. If I check off one item in the Tasklist HUD, and then go to a different one to check that one, it doesn't check it on the first click.  I have to click it twice.  If I check/uncheck the same item, however, it only requires a single click.  I think this is because on the first item that gets checked/unchecked, the focus changes to the editor.  When I then click on a different item in the HUD, it only puts the HUD into focus, but doesn't check/uncheck the box.  It thus requires a second mouse click to actually check/uncheck the desired item.
3. Sometimes when I check/uncheck a box for an item in the  HUD using the mouse, it's checking/unchecking the wrong task item.  I think this is somehow related to item #2 above.  I'm just not sure how.

In short, I recommend more testing for your code before I accept a pull request.  As a pro software engineer, might I suggest how you might in general go about this?  Pretend you are an evil hacker who wants to break the application.  Think sinisterly.  Break it.  Be ruthless.  Think, "What happens if I abuse it in these combinations?"  Write up a test plan--just a list of different combinations of things you can do to break it.  Test edge cases, such as what if you delete or add an item to the beginning of the list, or to the document?  What happens if it's at the end?  What about the middle of the list?  What happens if you have 300 task items?  What if you delete 100 of them?  What if you complete and check 50 of them?  What if some of them are indented in the list?

I hope that helps.

**Comment by hirschenberger** [comment]
@wereturtle Sorry, for me this PR ran out of steam. This process is not *agile* enough for me to wait another month for a reply. (I know, everybody is busy, time is money...)
This feature works well enough for me, to work on my novel. Feel free to fix the code or drop the PR.

Nevertheless, thank you for this great editor.

**Comment by wereturtle** [comment]
@hirschenberger No worries!  I do apologize for my lack of turnaround time.  Yes, this editor is also a procrastination tool for my own writing.  (My response time is slowing down because I've been trying to procrastinate less.)  I totally understand where you're coming from.  Good luck with your writing!
